core/src/main/java/org/apache/iceberg/GenericManifestEntry.java

```diff
@@ -44,14 +45,16 @@ private GenericManifestEntry(GenericManifestEntry<F> toCopy, boolean fullCopy) {
     this.schema = toCopy.schema;
     this.status = toCopy.status;
     this.snapshotId = toCopy.snapshotId;
+    this.dataSequenceNumber = toCopy.dataSequenceNumber;
     this.sequenceNumber = toCopy.sequenceNumber;
     this.file = toCopy.file().copy(fullCopy);
   }
 
-  ManifestEntry<F> wrapExisting(Long newSnapshotId, Long newSequenceNumber, F newFile) {
+  ManifestEntry<F> wrapExisting(Long newSnapshotId, Long newDataSequenceNumber, F newFile) {
```

Comment:
Nit: it would be fewer changes if we didn't rename `newSequenceNumber` to `newDataSequenceNumber` but I can see why you'd want to.

Reply:
Let me think. I wanted to be explicit in this case as we will also add `fileSequenceNumber` next.

Reply:
I decided to go with the rename as we have two variables now and I plan to add `fileSequenceNumber` next.

Reply:
@aokolnychyi , What class do you plan to add `fileSequenceNumber`? `DataFile` and `DeleteFile`?

Reply:
@chenjunjiedada, I'd say we add it to ManifestEntry first (to store as part of manifest entry) and then we can propagate it via `ContentFile`.

Reply:
@aokolnychyi, Thanks for the explanation, that would be very helpful.

Reply:
@chenjunjiedada, I recall you had a PR for propagating seq numbers to `ContentFile`. I'll reach out once this part is done so that you can rebase your work.

Reply:
@aokolnychyi , Yeap,  actually I have two PRs regarding sequence number, will rebase them when you finish these.